### PR TITLE
Feat/activity page

### LINKED ISSUE #117: Feat: Activity Page
Build out this activitity page that will show the upcoming and previous bookings for a guardian's children

add the nav bar icon for the flow

each activity should show which child it is for

Image

should have filter by child
Image

Image

### PR DESCRIPTION
**Pull request by bpalazzi512**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/117)
Added activity page to show upcoming/past registrations and be able to leave reviews on past registrations.

# How Has This Been Tested?
manually tested on ios and android

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)
Sent screen recording via slack

# Checklist

## General
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 1]
```diff
@@ -0,0 +1,761 @@
...
+interface RegistrationCardProps {
+  data: RegistrationCardData
+}
+
+const formatTime = (d: Date) =>
+  d.toLocaleTimeString([], { hour: "2-digit", minute: "2-digit" })
+const formatDate = (d: Date) =>
+  d.toLocaleDateString([], { weekday: "short", month: "short", day: "numeric" })
+
+const CHILD_BUBBLE_COLORS = [
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: These should be exported constants somewhere

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 2]
```diff
@@ -0,0 +1,761 @@
...
+      className="w-11/12 rounded-xl overflow-hidden mb-4"
+      style={{
+        borderWidth: 1,
+        borderColor: AppColors.borderLight,
+        backgroundColor: AppColors.white,
+        shadowColor: "#000",
+        shadowOffset: { width: 0, height: 1 },
+        shadowOpacity: 0.06,
+        shadowRadius: 4,
+        elevation: 2,
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: Nit style prop

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 3]
```diff
@@ -0,0 +1,761 @@
...
+const CHILD_BUBBLE_COLORS = [
+  "#E53E3E", "#DD6B20", "#D69E2E", "#276749", "#2B6CB0",
+  "#553C9A", "#B83280", "#00695C", "#C53030", "#2C5282",
+]
+
+function getInitials(name: string) {
+  return name.split(" ").map((n) => n[0]).join("").toUpperCase().slice(0, 2)
+}
+
+function UpcomingRegistrationCard({ data }: RegistrationCardProps) {
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: Can you split this out into its own component

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 4]
```diff
@@ -0,0 +1,761 @@
...
+          activeOpacity={0.7}
+        >
+          <ThemedText lightColor="white" className="text-sm font-medium">{t("activity.remove")}</ThemedText>
+        </TouchableOpacity>
+      </View>
+    </View>
+  );
+}
+
+function PastRegistrationCard({ data }: RegistrationCardProps) {
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: component

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 5]
```diff
@@ -0,0 +1,761 @@
...
+}
+
+export default function ActivityScreen() {
+  const insets = useSafeAreaInsets();
+  const router = useRouter();
+  const { t } = useTranslation();
+  const [selection, setSelection] = useState<"upcoming" | "past">("upcoming");
+
+  const { guardianId } = useAuthContext();
+  // const guardianId = "11111111-1111-1111-1111-111111111111";
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: Remove comment

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 6]
```diff
@@ -0,0 +1,761 @@
...
+          setCancelTarget(null)
+        } else {
+          Animated.spring(cancelSheetTranslateY, { toValue: 0, useNativeDriver: true }).start()
+        }
+      },
+    })
+  ).current
+
+  const getOnRemove = (childRegistrations: ChildRegistration[]) => {
+    return function () {
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: Nit can we use () =>

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 7]
```diff
@@ -0,0 +1,761 @@
...
+            <ThemedText lightColor="white" className="text-sm font-medium">{t("activity.review")}</ThemedText>
+          </TouchableOpacity>
+        )}
+      </View>
+    </View>
+  );
+}
+
+export default function ActivityScreen() {
+  const insets = useSafeAreaInsets();
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx:51]: For this hook can we move the logic and the hooks that make sense into a custom hook? It seems like we are trying to do a lot all in one component

File: frontend/apps/mobile/babel.config.js [hunk 8]
```diff
@@ -2,6 +2,16 @@ module.exports = function (api) {
   api.cache(true);
   return {
     presets: ["babel-preset-expo", "nativewind/babel"],
-    plugins: [],
+    plugins: [
```
Comment (josh-torre) [inline frontend/apps/mobile/babel.config.js:5]: Reason for these changes?

File: frontend/apps/mobile/app/(app)/(tabs)/activity/[id]/reviews.tsx
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/[id]/reviews.tsx:1]: Do we need this as a page? i feel like it would be better to just route to the existing reviews page for an event

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 9]
```diff
@@ -0,0 +1,761 @@
...
+  d.toLocaleTimeString([], { hour: "2-digit", minute: "2-digit" })
+const formatDate = (d: Date) =>
+  d.toLocaleDateString([], { weekday: "short", month: "short", day: "numeric" })
+
+const CHILD_BUBBLE_COLORS = [
+  "#E53E3E", "#DD6B20", "#D69E2E", "#276749", "#2B6CB0",
+  "#553C9A", "#B83280", "#00695C", "#C53030", "#2C5282",
+]
+
+function getInitials(name: string) {
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: we should be showing the child avatar instead of the initials

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 10]
```diff
@@ -0,0 +1,761 @@
...
+          <ThemedText type="subtitle" className="font-bold leading-none mr-1"> {data.start_time.getDate() < 10 ? "0" + data.start_time.getDate().toString() : data.start_time.getDate().toString()}</ThemedText>
+          <ThemedText type="subtitle" className=" font-semibold tracking-widest "> {data.start_time.toLocaleString('default', { month: 'short' }) }</ThemedText>
+        </View>
+      </View>
+      <View
+        className="flex flex-row justify-between items-center px-4 py-3"
+      >
+        <View className="flex flex-row gap-2">
+          {children.map((child) => {
+            const color = data.childColorMap[child.id] ?? CHILD_BUBBLE_COLORS[0]
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: child avatars will include the color also so this wont be needed

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 11]
```diff
@@ -0,0 +1,761 @@
...
+        </TouchableOpacity>
+      </View>
+    </View>
+  );
+}
+
+function PastRegistrationCard({ data }: RegistrationCardProps) {
+  const { t } = useTranslation();
+  const children = data.childRegistrations.map((cr) => cr.child)
+  const priceDisplay = `฿${(data.price / 100).toLocaleString()}`
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: why divide by 100?

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx [hunk 12]
```diff
@@ -0,0 +1,761 @@
...
+    guardianId!,
+    {
+      query: { enabled: !!guardianId },
+    },
+  );
+  const registrations: Registration[] = useMemo(() => {
+    const d = registrationsResp as unknown as
+      | { data: { registrations: Registration[] } }
+      | undefined;
+    // console.log(d?.data?.registrations)
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: remove comment

File: frontend/apps/mobile/babel.config.js (hunk 8 above)
Comment (bpalazzi512) [reply frontend/apps/mobile/babel.config.js:5]: Was getting errors without it

File: frontend/apps/mobile/app/(app)/(tabs)/activity/[id]/reviews.tsx
Comment (bpalazzi512) [reply frontend/apps/mobile/app/(app)/(tabs)/activity/[id]/reviews.tsx:1]: This allows us to go back to the previous page in a smooth way

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx (hunk 11 above)
Comment (bpalazzi512) [reply frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx]: price is in cents

File: frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx (hunk 7 above)
Comment (bpalazzi512) [reply frontend/apps/mobile/app/(app)/(tabs)/activity/index.tsx:51]: Just separated out the data grabbing part into a hook, lmk if you want further separation

**Review by adescoteaux1 — DISMISSED** [review]
LGTM 🤘

**Review by adescoteaux1 — APPROVED** [review]
